### PR TITLE
[#166734445] Remove Services from Static Container

### PR DESCRIPTION
**Pull request by BurnedMarshal**
Remove all Services initialization from Static Container. Rename container file to config.
This PR is dependent from #470 . When the master PR is merged change the base branch to `166734445-remove-awilix`

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#166734445](https://www.pivotaltracker.com/story/show/166734445): Rimuovere awilix dal codice

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/app.ts [hunk 1]
```diff
@@ -217,6 +216,16 @@ export async function newApp(
     defaultModule.currentSpidStrategy = await generateSpidStrategy();
     defaultModule.registerLoginRoute(app, defaultModule.currentSpidStrategy);
     registerPublicRoutes(app);
+
+    // Create the Session Storage ervice
```
Comment (cloudify) [inline src/app.ts]:
```suggestion
    // Create the Session Storage service
```

File: src/config.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+import { getRequiredENVVar, readFile } from "./utils/container";
+import { log } from "./utils/logger";
+
+import {
+  createClusterRedisClient,
+  createSimpleRedisClient
+} from "./utils/redis";
+
+// Without this the environment variables loaded by dotenv aren't available in
```
Comment (cloudify) [inline src/config.ts]:
```suggestion
// Without this, the environment variables loaded by dotenv aren't available in
```

File: src/config.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+// Redirection urls
+const clientProfileRedirectionUrl =
+  process.env.CLIENT_REDIRECTION_URL || "/profile.html?token={token}";
+
+if (!clientProfileRedirectionUrl.includes("{token}")) {
+  log.error("CLIENT_REDIRECTION_URL must contains a {token} placeholder");
+}
+
+// Range IP allowed for notification.
+function decodeNotifyCIDR(): string & IPatternStringTag<string> {
```
Comment (cloudify) [inline src/config.ts]:
```suggestion
function decodeNotifyCIDR(): CIDR {
```

https://github.com/teamdigitale/io-ts-commons/blob/master/src/strings.ts#L129

File: src/config.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    );
+    return process.exit(1);
+  } else {
+    return errorOrNotifyCIDR.value;
+  }
+}
+export const ALLOW_NOTIFY_IP_SOURCE_RANGE = decodeNotifyCIDR();
+
+// Range IP allowed for PagoPA proxy.
+function decodePagoPACIDR(): string & IPatternStringTag<string> {
```
Comment (cloudify) [inline src/config.ts]:
```suggestion
function decodePagoPACIDR(): CIDR {
```

File: src/config.ts [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+      ReadableReporter.report(errorOrPagoPACIDR)
+    );
+    return process.exit(1);
+  } else {
+    return errorOrPagoPACIDR.value;
+  }
+}
+export const ALLOW_PAGOPA_IP_SOURCE_RANGE = decodePagoPACIDR();
+
+export const API_KEY = process.env.API_KEY || "";
```
Comment (cloudify) [inline src/config.ts]: these are no sensible defaults, server should not start without these

File: src/config.ts [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+// Set default session duration to 30 days
+const DEFAULT_TOKEN_DURATION_IN_SECONDS = 3600 * 24 * 30;
+export const tokenDurationSecs: number = process.env.TOKEN_DURATION_IN_SECONDS
+  ? parseInt(process.env.TOKEN_DURATION_IN_SECONDS, 10)
+  : DEFAULT_TOKEN_DURATION_IN_SECONDS;
+log.info("Session token duration set to %s seconds", tokenDurationSecs);
+
+// Register a factory service to create PagoPA client.
+const pagoPAApiUrl = process.env.PAGOPA_API_URL || "";
```
Comment (cloudify) [inline src/config.ts]: make it required

File: src/config.ts [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+const pagoPAApiUrlTest = process.env.PAGOPA_API_URL_TEST || "";
```
Comment (cloudify) [inline src/config.ts]: make it required

File: src/app.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/app.ts]: Fixed in 7db18a3

File: src/config.ts (hunk 5 above)
Comment (BurnedMarshal) [reply src/config.ts]: Fixed in 7db18a3

File: src/strategies/bearerTokenStrategy.ts [hunk 8]
```diff
@@ -28,7 +29,10 @@ const bearerTokenStrategy = (
     done: (error: any, user?: any, options?: IVerifyOptions | string) => void
   ) => {
     const path = req.route.path;
-    const sessionStorage: ISessionStorage = SESSION_STORAGE;
+    const sessionStorage: ISessionStorage = new RedisSessionStorage(
```
Comment (gunzip) [inline src/strategies/bearerTokenStrategy.ts]: that would be better to pass the sessionstorage as a param, see:

https://github.com/AgID/agid-forms-backend/blob/master/src/strategies/bearer_token.ts

Comment (BurnedMarshal) [reply src/strategies/bearerTokenStrategy.ts]: Fixed in 3e0d913

**Pull request merged**